alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/model/destination/email/EmailAccount.kt

```diff
@@ -0,0 +1,134 @@
+        builder.field(SCHEMA_VERSION, schemaVersion)
+                .field(NAME_FIELD, name)
+                .field(EMAIL_FIELD, email)
+                .field(HOST_FIELD, host)
+                .field(PORT_FIELD, port)
+                .field(METHOD_FIELD, method.value)
+        if (params.paramAsBoolean("with_type", false)) builder.endObject()
+        return builder.endObject()
+    }
+
+    fun toXContent(builder: XContentBuilder): XContentBuilder {
+        return toXContent(builder, ToXContent.EMPTY_PARAMS)
+    }
+
+    enum class MethodType(val value: String) {
+        NONE("none"),
+        SSL("ssl"),
+        TLS("starttls");
+
+        companion object {
+            private val values = values()
+            // Created this method since MethodType value does not necessarily match enum name
+            fun getByValue(value: String) = values.firstOrNull { it.value == value }
+        }
+    }
+
+    companion object {
+        const val EMAIL_ACCOUNT_TYPE = "email_account"
+        const val NO_ID = ""
+        const val SCHEMA_VERSION = "schema_version"
+        const val NAME_FIELD = "name"
+        const val EMAIL_FIELD = "email"
+        const val HOST_FIELD = "host"
+        const val PORT_FIELD = "port"
+        const val METHOD_FIELD = "method"
+
+        @JvmStatic
+        @Throws(IOException::class)
+        fun parse(xcp: XContentParser, id: String = NO_ID): EmailAccount {
+            var schemaVersion = NO_SCHEMA_VERSION
+            lateinit var name: String
+            lateinit var email: String
+            lateinit var host: String
+            var port: Int = -1
+            lateinit var method: String
+
+            ensureExpectedToken(Token.START_OBJECT, xcp.currentToken(), xcp::getTokenLocation)
+            while (xcp.nextToken() != Token.END_OBJECT) {
+                val fieldName = xcp.currentName()
+                xcp.nextToken()
+
+                when (fieldName) {
+                    SCHEMA_VERSION -> schemaVersion = xcp.intValue()
+                    NAME_FIELD -> name = xcp.text()
+                    EMAIL_FIELD -> email = xcp.text()
+                    HOST_FIELD -> host = xcp.text()
+                    PORT_FIELD -> port = xcp.intValue()
+                    METHOD_FIELD -> {
+                        method = xcp.text()
+                        val allowedMethods = MethodType.values().map { it.value }
+                        if (!allowedMethods.contains(method)) {
+                            throw IllegalStateException("Method should be one of $allowedMethods")
+                        }
+                    }
+                }
+            }
+
+            return EmailAccount(id,
+                    schemaVersion,
+                    name,
+                    email,
+                    host,
+                    port,
+                    requireNotNull(MethodType.getByValue(method)) { "Method type was null" }
```

Comment:
Nitpick: I prefer if this `requireNotNull` check and value setting is done above and set into a variable that will be passed into the constructor.

Reply:
This is actually a conventional use of `requireNotNull` since we've already established the parameter is not nullable but the parse value can be null, we check it here with a custom lazyMessage instead of adding another variable. You can see similar examples in other data classes.